lib/dictBuilder/cover.c

```diff
@@ -929,6 +934,111 @@ void COVER_best_finish(COVER_best_t *best, size_t compressedSize,
   }
 }
 
+COVER_dictSelection_t COVER_dictSelectionError(size_t error) {
+    COVER_dictSelection_t selection = { NULL, error, error };
```

Comment:
You need to set `dictSize == 0` here.

In `COVER_best_finish()` when `totalCompressedSize` is an error, but still `< -1` we will allocate space for the dictionary. We will try to allocate ~2^64 bytes, which will probably fail, but still we don't want to do that.

The dictionary size should always be the right number here, which is `0`.